docs/cross-contract/callbacks.md

```diff
@@ -46,61 +46,125 @@ mod ext_calculator {
 
-The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up as interfaces for the smart contract being called, and (typically) the current contract doing the calling (where callback logic will live). The second trait is, by convention, referred to as `ext_self`. Here are the two traits used in a simple example:
+The following example demonstrates two common approaches to callbacks using the high-level cross-contract approach. When writing high-level cross-contract calls, special [traits](https://doc.rust-lang.org/rust-by-example/trait.html) are set up as interfaces for the smart contract being called.
 
-```rust reference
-https://github.com/mikedotexe/cross-contract-view/blob/06bdb3222622c824b9f2fe0a53536e6914435580/src/lib.rs#L17-L26
+```rust
+#[ext_contract(ext_allowlist)]
+pub trait ExtAllowlist {
+    fn is_allowlisted(staking_pool_account_id: AccountId) -> bool;
+}
 ```
 
-After creating these traits, we'll show two simple functions that will make a cross-contract call to a whitelist smart contract, asking if the account `mike.testnet` is whitelisted. These methods will both return `true` using different approaches. First we'll look at the methods, then we'll look at the differences in callbacks. Note that for simplicity in this example, the values are hardcoded.
+After creating the trait, we'll show two simple functions that will make a cross-contract call to an allowlist smart contract, asking if the account `mike.testnet` is allowlisted. These methods will both return `true` using different approaches. First we'll look at the methods, then we'll look at the differences in callbacks. Note that for simplicity in this example, the values are hardcoded.
 
 ```rust
-pub const XCC_GAS: Gas = 20000000000000;
-fn get_whitelist_contract() -> AccountId {
-    "whitelist.demo.testnet".to_string()
+pub const XCC_GAS: Gas = Gas(20000000000000);
+fn get_allowlist_contract() -> AccountId {
+    "allowlist.demo.testnet".parse().unwrap()
 }
 fn get_account_to_check() -> AccountId {
-    "mike.testnet".to_string()
+    "mike.testnet".parse().unwrap()
 }
 ```
 
-```rust reference
-https://github.com/mikedotexe/cross-contract-view/blob/06bdb3222622c824b9f2fe0a53536e6914435580/src/lib.rs#L32-L52
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn xcc_use_promise_result(&mut self) -> Promise {
+        // Call the method `is_allowlisted` on the allowlisted contract. Attach static GAS equal to XCC_GAS.
+        // Any unused GAS will be split between the function call and the callback since both have a default unused GAS weight of 1
+        // Attached deposit is defaulted to 0 for both the function call and the callback.
+        ext_allowlist::ext(get_allowlist_contract())
+            .with_static_gas(XCC_GAS)
```

Comment:
static gas probably isn't required for the first part. If this fails, there is no intermediate state changes

Reply:
Fixed